src/crafting.cpp

```diff
@@ -164,6 +164,16 @@ void player::long_craft()
     }
 }
 
+void player::show_craft( const recipe *filter )
+{
+    hidden_recipes->remove( filter );
```

Comment:
`hidden_recipes` is a solely GUI-related thing. It should probably live elsewhere (outside the `player` class).

Reply:
Elsewhere like where? Considering it needs to be saved.

Reply:
Like for example in `uistatedata`. `player` is already an uber-bloated class. It would be better not to festoon it even more with unrelated stuff. Especially considering that `nps` is a descendant of player (which is not a good thing, but so it is for now).